core/src/target_os_check.rs

```diff
@@ -0,0 +1,329 @@
+    fn test_target_os_combined_any_accepted() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(any(target_os = "android", target_os = "ios"))]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_target_os_combined_all_accepted() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(all(target_os = "windows", target_os = "android"))]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_target_os_combined_rejected() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(not(any(target_os = "wasm32", target_os = "ios")))]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(!accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_accept_no_target_os() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(feature = "my-feature")]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_accept_no_attribute() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            pub struct NestedNotTarget;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_accept_none_excluded() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(not(any(target_os = "wasm32", target_os = "ios")))]
+            pub struct Excluded;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["macos".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_reject_not_target_os() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(target_os = "ios")]
+            pub struct Excluded;
+        };
+
+        assert!(!accept_target_os(
+            &test_struct.attrs,
+            &["macos".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_any_target_not_target_os() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(any(target_os = "ios", feature = "test"))]
+            pub struct Excluded;
+        };
+
+        assert!(!accept_target_os(
+            &test_struct.attrs,
+            &["macos".into(), "android".into()]
+        ));
+    }
```

Comment:
For our own benefit in the future, as well as the benefit of new users of this feature, is it too much of a burden to write a section of the docs about how we process the `target_os` directives? This test is an example that feels, on the surface, incorrect—if you compile the code with `--features test`, even on `macos`, the struct `Excluded` will actually be included. So whether the `target_os` matters or not depends on the feature set.

I absolutely believe that trying to fully recreate the `cfg` parser is beyond the scope of this feature, however since we are effectively ignoring any other directives beyond the `target_os` ones, we should make that clear so when we run into a question about that in the future we have a clear path to understanding why the behavior is the way it is.

Reply:
Sure. I added `target_os.md` to the typeshare book to describe how the feature works. I'm somewhat exploring what other use cases others may require. I had an initial use case, however after exposing others to this feature, additional uses cases came up which evolved into this PR.